src/test/java/teammates/test/cases/webapi/UpdateFeedbackResponseActionTest.java

```diff
@@ -158,7 +215,273 @@ public void testExecute_instructorFeedbackSubmissionMcqGenerateOptionsForTeams_s
+        };
+
+        UpdateFeedbackResponseAction a = getAction(updateRequest, moderatedStudentSubmissionParams);
+        assertThrows(UnauthorizedAccessException.class, a::checkAccessControl);
+    }
+
+    @Test
+    protected void testAccessControl_studentAccessOtherStudent_inaccessible() throws Exception {
+
+        ______TS("Student intends to access other person's response, should not be accessible");
+
+        loginAsStudent(student2InCourse1.getGoogleId());
+
+        assertEquals(FeedbackParticipantType.STUDENTS,
+                logic.getFeedbackQuestion(typicalResponse.getFeedbackQuestionId()).getGiverType());
+        assertNotEquals(student2InCourse1.getEmail(), typicalResponse.getGiver());
+
+        // send update request
+        FeedbackResponseUpdateRequest updateRequest = new FeedbackResponseUpdateRequest();
+        updateRequest.setQuestionType(FeedbackQuestionType.MCQ);
+        updateRequest.setRecipientIdentifier(typicalResponse.getRecipient());
+        FeedbackMcqResponseDetails newDetails = new FeedbackMcqResponseDetails();
+        newDetails.setAnswer("TEAM_NOT_EXIST");
+        updateRequest.setResponseDetails(newDetails);
+
+        String[] studentAccessOtherPersonParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.STUDENT_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, typicalResponse.getId(),
+        };
+
+        UpdateFeedbackResponseAction a = getAction(updateRequest, studentAccessOtherPersonParams);
+        assertThrows(UnauthorizedAccessException.class, a::checkAccessControl);
+    }
+
+    @Test
+    protected void testAccessControl_studentAccessOtherTeam_inaccessible() {
+
+        ______TS("Student intends to access other team's response, should not be accessible");
+
+        loginAsStudent(student5InCourse1.getGoogleId());
+
+        assertEquals(FeedbackParticipantType.TEAMS,
+                logic.getFeedbackQuestion(typicalResponse2.getFeedbackQuestionId()).getGiverType());
+        assertNotEquals(student5InCourse1.getTeam(), typicalResponse2.getGiver());
+
+        // send update request
+        FeedbackResponseUpdateRequest updateRequest = new FeedbackResponseUpdateRequest();
+        updateRequest.setQuestionType(FeedbackQuestionType.MCQ);
+        updateRequest.setRecipientIdentifier(student5InCourse1.getEmail());
+        FeedbackMcqResponseDetails newDetails = new FeedbackMcqResponseDetails();
+        newDetails.setAnswer("TEAM_NOT_EXIST");
+        updateRequest.setResponseDetails(newDetails);
+
+        String[] studentAccessOtherTeamParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.STUDENT_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, typicalResponse2.getId(),
+        };
+
+        UpdateFeedbackResponseAction a = getAction(updateRequest, studentAccessOtherTeamParams);
+        assertThrows(UnauthorizedAccessException.class, a::checkAccessControl);
```

Comment:
verifyCannotAccess(studentAccessOtherTeamParams)

Reply:
I did this initially because `verifyCannotAccess()` can't take in an `UpdateRequest` as a parameter. But now that I think about it I could overload the method thanks for the inspiration! :)